### PR TITLE
[MM-37108] Only make request if server is enterprise ready

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Making the `getPrevTrialLicense()` request was causing a panic server side when in team edition. 

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-37108

#### Release Note

```release-note
NONE
```

#### Related PRs

https://github.com/mattermost/mattermost-server/pull/17933

### CODE REVIEW

**Review by harshilsharma63 — APPROVED** [review]

File: components/admin_console/license_settings/license_settings.jsx [hunk 1]
```diff
@@ -71,7 +71,9 @@ export default class LicenseSettings extends React.PureComponent {
             this.reloadPercentage();
         }
         this.props.actions.getLicenseConfig();
-        this.props.actions.getPrevTrialLicense();
+        if (this.props.enterpriseReady) {
+            this.props.actions.getPrevTrialLicense();
+        }
```
Comment (streamer45) [inline components/admin_console/license_settings/license_settings.jsx:76]: If the order of these calls is not important I can just add the call to an else block. Let me know.

Comment (BenCookie95) [reply components/admin_console/license_settings/license_settings.jsx:76]: I don't think the order of these calls matter, you could move it into the else if you want

**Review by BenCookie95 — APPROVED** [review]

**Review by furqanmlk — APPROVED** [review]
Working as expected
Details are updated in Server PR

**Pull request merged**